analytical_engine/core/fragment/dynamic_fragment.h

```diff
@@ -2417,6 +2445,8 @@ class DynamicFragment {
 
   template <typename _VDATA_T, typename _EDATA_T>
   friend class DynamicProjectedFragment;
+
+  friend class DynamicFragmentView;
```

Comment:
If the DynamicFragment is intended to be inherited, it's better to qualify these methods like inner_ie_pos(), inner_edge_space() with the protected qualifier, or turn methods into attributes.